prow/githubeventserver/githubeventserver.go

```diff
@@ -0,0 +1,495 @@
+					github.OrgLogField:  pr.Repo.Owner.Login,
+					github.RepoLogField: pr.Repo.Name,
+					github.PrLogField:   pr.Number,
+					"author":            pr.PullRequest.User.Login,
+					"url":               pr.PullRequest.HTMLURL,
+				}), pr)
+			}()
+		}
+
+	case pullRequestReviewEvent:
+		var re github.ReviewEvent
+		if err := json.Unmarshal(payload, &re); err != nil {
+			return err
+		}
+		re.GUID = eventGUID
+		org = re.Repo.Owner.Login
+		repo = re.Repo.Name
+
+		for _, reviewEventHandler := range s.reviewEventHandlers {
+			fn := reviewEventHandler
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
+				fn(l.WithFields(logrus.Fields{
+					github.OrgLogField:  re.Repo.Owner.Login,
+					github.RepoLogField: re.Repo.Name,
+					github.PrLogField:   re.PullRequest.Number,
+					"review":            re.Review.ID,
+					"reviewer":          re.Review.User.Login,
+					"url":               re.Review.HTMLURL,
+				}), re)
+			}()
+		}
+
+	case pullRequestReviewCommentEvent:
+		var rce github.ReviewCommentEvent
+		if err := json.Unmarshal(payload, &rce); err != nil {
+			return err
+		}
+		rce.GUID = eventGUID
+		org = rce.Repo.Owner.Login
+		repo = rce.Repo.Name
+
+		for _, reviewCommentEventHandler := range s.reviewCommentEventHandlers {
+			fn := reviewCommentEventHandler
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
+				fn(l.WithFields(logrus.Fields{
+					github.OrgLogField:  rce.Repo.Owner.Login,
+					github.RepoLogField: rce.Repo.Name,
+					github.PrLogField:   rce.PullRequest.Number,
+					"review":            rce.Comment.ReviewID,
+					"commenter":         rce.Comment.User.Login,
+					"url":               rce.Comment.HTMLURL,
+				}), rce)
+			}()
+		}
+
+	case pushEvent:
+		var pe github.PushEvent
+		if err := json.Unmarshal(payload, &pe); err != nil {
+			return err
+		}
+		pe.GUID = eventGUID
+		org = pe.Repo.Owner.Login
+		repo = pe.Repo.Name
+
+		for _, pushEventHandler := range s.pushEventHandlers {
+			fn := pushEventHandler
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
+				fn(l.WithFields(logrus.Fields{
+					github.OrgLogField:  pe.Repo.Owner.Name,
+					github.RepoLogField: pe.Repo.Name,
+					"ref":               pe.Ref,
+					"head":              pe.After,
+				}), pe)
+			}()
+		}
+
+	case statusEvent:
+		var se github.StatusEvent
+		if err := json.Unmarshal(payload, &se); err != nil {
+			return err
+		}
+		se.GUID = eventGUID
+		org = se.Repo.Owner.Login
+		repo = se.Repo.Name
+
+		for _, statusEventHandler := range s.statusEventHandlers {
+			fn := statusEventHandler
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
+				fn(l.WithFields(logrus.Fields{
+					github.OrgLogField:  se.Repo.Owner.Login,
+					github.RepoLogField: se.Repo.Name,
+					"context":           se.Context,
+					"sha":               se.SHA,
+					"state":             se.State,
+					"id":                se.ID,
+				}), se)
+			}()
+		}
+
+	default:
+		l.Debug("Ignoring unhandled event type.")
```

Comment:
Nit: Log the event type?

Reply:
This logger already holds the event type in its fields. See line 209